crates/miden-lib/asm/kernels/transaction/lib/prologue.masm

```diff
@@ -461,13 +461,30 @@ proc.process_account_data
     exec.memory::get_current_account_data_ptr
     # => [acct_data_ptr]
 
-    # read account details and compute its digest. See `Advice stack` above for details.
-    padw padw padw
-    adv_pipe hperm
-    adv_pipe hperm
+    # Prepare the stack for the account commitment computation. See `Advice stack` above for more 
+    # details.
+    padw padw padw adv_pipe
+    # => [ACCOUNT_VAULT_ROOT, ID_AND_NONCE, EMPTY_WORD, acct_data_ptr']
+
+    # store the initial vault root to the memory
+    exec.memory::set_init_account_vault_root
+    # => [ACCOUNT_VAULT_ROOT, ID_AND_NONCE, EMPTY_WORD, acct_data_ptr']
+    
+    # perform the permutation and load storage and code commitments onto the operand stack 
+    hperm adv_pipe
+    # => [ACCOUNT_CODE_COMMITMENT, ACCOUNT_STORAGE_COMMITMENT, HASH, acct_data_ptr'']
+
+    # store the initial code and storage commitments
+    exec.memory::set_init_account_code_commitment swapw
+    exec.memory::set_init_account_storage_commitment swapw
```

Comment:
Question: why are we "breaking up" the hashing process here to save the data rather than saving it a bit further down in this function (similar to how we save account nonce on line 499 below)?

For example, instead of calling `set_init_account_storage_commitment` we could just call it on line 504 below. I think this would be more consistent with how we currently handle account nonce.

This also applies to account code and vault commitments.

Reply:
The idea was to use the commitments while they are still on the stack to avoid the extra call of the commitment getter procedure to save a few cycles. But I missed the fact that we are getting this exact values down below to save the storage data, new code commitment and an input vault root, so it will be indeed better not to break up the hashing process and just use these values later in this procedure.